Sources/Common/CustomerIOInstance.swift

```diff
@@ -10,15 +10,6 @@ public protocol CustomerIOInstance: AutoMockable {
         body: [String: Any]
     )
 
-    // sourcery:Name=identifyEncodable
-    // sourcery:DuplicateMethod=identify
-    func identify<RequestBody: Encodable>(
-        identifier: String,
-        // sourcery:Type=AnyEncodable
-        // sourcery:TypeCast="AnyEncodable(body)"
-        body: RequestBody
-    )
-
```

Comment:
(this comment can be applied to all code blocks removed in this PR, not just this one)

I don't understand the reason to remove these functions instead of keeping them and changing the function signature? 

Has this been considered as a solution? 

```swift
    // change Encodable to Codable
    func identify<RequestBody: Codable>(
```

Reply:
Update: I re-read the description a few times and what I think you're suggesting is that the CDP's extension methods that it has added will allow function signatures such as 
```swift
func identify<RequestBody: Codable>(
```

to be added. Therefore, the public API will still allow `func identify<RequestBody: Codable>(` to exist in the SDK, but you will need to import the CDP module in your Swift files to use it. 

Assuming that is all correct, I will continue to ask my original question of: has it been considered to keep all of these functions in the Common protocol? It doesn't feel right to me to have some forms of identify/track functions available in Common module while having more non-Segment specific identify/track functions available in another module.

Reply:
That is a very fair demand, for method that doesn't involve Segment oriented classes, customers shouldn't have to add external dependency for those. 

> keep all of these functions in the Common protocol?

But we won't want to keep *all* right? because first `common` module doesn't have any information about the `CDP` module, secondly, we won't want to bound `common` module with segment specific classes like `plugin` so we can easily deviate when needed.

Reply:
Another [PR](https://github.com/customerio/customerio-ios/pull/421) is made which updates the public method.